src/libsync/syncengine.cpp

```diff
@@ -1179,6 +1180,34 @@ void SyncEngine::checkForPermission(SyncFileItemVector &syncItems)
             if ((*it)->_isDirectory) {
                 for (SyncFileItemVector::iterator it_next = it + 1; it_next != syncItems.end() && (*it_next)->_file.startsWith(path); ++it_next) {
                     it = it_next;
+                    if ((*it)->_direction != SyncFileItem::Up && (*it)->_instruction == CSYNC_INSTRUCTION_REMOVE) {
```

Comment:
Why `!= Up` and not `== Down`? I'd be more comfortable with this pointing specifically at local removes.